sandy/core/cov.py

```diff
@@ -429,12 +427,12 @@ def _reduce_size(self):
         >>> equal_arrays = comparison.all()
         >>> assert equal_arrays == True
         >>> reduce_matrix
-                      1	          2
-        1	2.00000e+00	0.00000e+00
-        2	0.00000e+00	3.00000e+00
+                    1           2
+        1 2.00000e+00 0.00000e+00
+        2 0.00000e+00 3.00000e+00
         """
         nonzero_idxs = np.flatnonzero(np.diag(self.data))
-        cov_reduced = self.data.loc[nonzero_idxs, nonzero_idxs]
+        cov_reduced = self.data.iloc[nonzero_idxs, nonzero_idxs]
         return nonzero_idxs, cov_reduced

```

Comment:
If you changed this it means that you found an example where `loc` fails and you decided that `iloc` should be used.
Can you add this test/example?

Reply:
The problem is that iloc identifies the position by integer index, while loc uses colunm label-based locationing and, according with the documentation, np.flatnonzero will return the _indices_ of the non-zero elements. This method with `loc` would work only if the columns labels were the same of the indexes (integers in range(df.shape[1])). I saw this problem performing a test with `Cov = sandy.XsCov.from_endf6(endf6)` in which the indeces don't coincide with the columns labels

Reply:
Ok! Then add an example that fails with loc